Sage/carbon
# fix(hidden-checkable-input): make inputref prop optional

Fixes: #4450

### Proposed behaviour

Make `inputRef` an optional prop in the `HiddenCheckableInput` TypeScript interface. This now matches the corresponding JavaScript propType.

### Current behaviour

`inputRef` was incorrectly marked as required.

![image](https://user-images.githubusercontent.com/14963680/136563833-76b9bbda-c721-4cd8-b9f9-b26b1ebbc89d.png)


### Checklist


#### QA


### Additional context

### Testing instructions
The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

https://codesandbox.io/s/goofy-cloud-zs3be?file=/src/index.tsx

The TypeScript sandboxes do not work in the browser unfortunately. Follow these steps to run it locally: 

- Click `Export to ZIP` on the codesandbox
- Unzip the folder in your downloads
- cd into the folder in a terminal and run `npm install`
- Look at the `src/index.tsx` file in vscode and you will no longer see the TypeScript error in the screenshot above on the `Checkbox` component.
